src/applicationLayer.c

```diff
@@ -0,0 +1,42 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "../include/utils.h"
+#include "../include/datenhaltung.h"
+#include "../include/applicationLayer.h"
+
+Result put(char* key, char* value ){
+    Result result;
+    if(save(key,value)==0){
+        result.value = value;
+        result.error_code = save(key, value);
+        return result;
+    }
+    else{
+        result.error_code = save(key, value);
+        return result;
+    }

```

Comment:
One call to save is sufficient. Moreover multiple calls introduces ambiguity, because the state might have changed in between.

With the new Result structure, the value string has a fixed length. So `strncpy` would be better than the assignment. But I guess this is alright for now.